### ISSUE TITLE
Add H264/H265/AV1 GPU accelerated video decode support

### ISSUE THREAD

**Issue by Ciancy28** [Genymobile/scrcpy#3800]
- [x] I have checked that a similar [feature request](https://github.com/Genymobile/scrcpy/issues?q=is%3Aopen+is%3Aissue+label%3A%22feature+request%22) does not already exist.

**Is your feature request related to a problem? Please describe.**
Currently, scrcpy doesen't use the GPU's hardware accelerated video decoding capabilities, which increases resource usage/decode delay, this doesen't impact most modern PCs much, but on systems with weaker processors (especially with AV1), it should make a big difference, as far as I know, there is no downside to using video decoding acceleration, and most applications already use it as expected (web browsers, telegram/whatsapp, video editing software, every media player in existance... scrcpy is the only application I've ever found that doesen't take advantage of this).   

**Describe the solution you'd like**
Hardware accelerated video decode should be either an option, or enabled by default and togglable with a parameter.   

**Describe alternatives you've considered**

**Additional context**
![H264 scrcpy](https://user-images.githubusercontent.com/77066322/224558842-d505a786-22fc-4d73-be95-883a1cf9ccb0.png)
![H264 movie](https://user-images.githubusercontent.com/77066322/224558646-8a52e5df-96cf-4620-8725-9bf47acddc90.png)
![H265 scrcpy](https://user-images.githubusercontent.com/77066322/224558651-51992ece-b405-4607-bfa0-dfee18dbf1e1.png)
![H265 movie](https://user-images.githubusercontent.com/77066322/224558932-eff6ab1f-96a4-43a2-a1fc-61a7e4d20ef7.png)

I observed the lack of gpu video decoding capabilities on a Windows 11 system with an AMD RX 580 GPU running scrcpy v2.0, I'm not sure if hardware decoding is limited to Linux or Nvidia/Intel GPUs, I couldn't find anything regarding this subject in the documentation, so I think it's missing on every system.
I couldn't test the presence of AV1 video decoding capabilities as I don't have an Android device with an AV1 encoder and I don't have a GPU with an AV1 decoder.

**Comment by rom1v** [comment]
Yes, that would be great. But that's not trivial to keep the decoded image on the GPU until display for all video output. On VLC, we implement interops manually for OpenGL, DirectX, etc. I have not the resource to develop and maintain all video outputs manually. I would like SDL to support it: https://github.com/libsdl-org/SDL/issues/5405

Refs https://github.com/Genymobile/scrcpy/pull/1894

> there is no downside to using video decoding acceleration

Now it's better than before, but hardware decoders caused a lot of crashes and other issues. In VLC, whenever such a problem occur, we recommend to disable hardware acceleration.

But of course when it works, it's better to use it.

**Comment by rom1v** [comment]
Oh, there are some news regarding this in SDL3: 
 - https://github.com/libsdl-org/SDL/issues/5405
 - https://github.com/libsdl-org/SDL/pull/8366

**Comment by oxwivi** [comment]
SDL is adding [Vulkan AV1 decode](https://github.com/libsdl-org/SDL/commit/d51b53ef8f76ee6dc49115fb7981b84e5b0f02b4). How difficult would it be to get scrcpy hooked into using those?

**Comment by DienoX** [comment]
@rom1v 
Thank you for wanting to port the project to SDL3 and provide GPU decode acceleration to this great project.

**Comment by DienoX** [comment]
Thanks to @rom1v  for migrating to SDL3 stable. I hope the next major 5.0 release will have GPU decoder acceleration.
